client/cody/src/chat/ChatViewProvider.ts

```diff
@@ -84,14 +84,37 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
         this.configurationChangeEvent.fire()
     }
 
-    public clearAndRestartSession(): void {
+    public async clearAndRestartSession(): Promise<void> {
+        await this.saveChatHistory()
         this.createNewChatID()
         this.cancelCompletion()
         this.transcript.reset()
         this.sendTranscript()
         this.sendChatHistory()
     }
 
+    public async clearHistory(): Promise<void> {
+        this.chatHistory = {}
+        this.inputHistory = []
+        await this.localStorage.removeChatHistory()
+    }
+
+    public async restoreSession(chatID: string): Promise<void> {
+        this.saveChatHistory()
+        this.cancelCompletion()
+
+        this.currentChatID = chatID
+        this.transcript = Transcript.fromJSON(this.chatHistory[chatID])
+        delete this.chatHistory[chatID]
+
+        this.sendTranscript()
+        await this.localStorage.setChatHistory({
```

Comment:
```suggestion
        await this.saveChatHistory()
```
Seems like this code is doing the same thing as `this.saveChatHistory()`?

Reply:
There's actually a subtle difference - `saveChatHistory` performs a `this.chatHistory[this.currentChatID] = await this.transcript.toJSON()` which I don't want as we don't want the current chat session to also be in the history. I can probably make that assignment its own function?